rust/headless-client/src/ipc_service.rs

```diff
@@ -124,7 +124,9 @@ fn run_smoke_test() -> Result<()> {
     crate::setup_stdout_logging()?;
     let rt = tokio::runtime::Runtime::new()?;
     let _guard = rt.enter();
-    let mut dns_controller = DnsController::default();
+    let mut dns_controller = DnsController {
+        dns_control_method: Default::default(),
+    };
```

Comment:
It appears as if we could stick with a `Default` impl here given the latest iteration?

Reply:
We could, but the smoke test is not part of the main code, and I don't want the main code to accidentally ignore the CLI args and construct a default `DnsController`. It should be difficult to construct one without thinking of which control method you want to use